collections/soldiers.js

```diff
@@ -0,0 +1,34 @@
+const MongoClient = require('mongodb').MongoClient;
+
+const url = 'mongodb://localhost:27017';
+const dbName = 'myproject';
+
+function checkData(soldierData) {
+  if (/tt[0-9]{7}/.test(soldierData["id"]) &&
+    soldierData["name"] &&
+    soldierData["rank"] &&
+    soldierData["limitations"] &&
+    Object.keys(soldierData).length === 5) {
```

Comment:
5 should be a const, defined at top of page or top of checkData method.
Also validate the last field `duties`.
Right now it's sort of ok because you've added it before the check - but if this method would be used elsewhere, it is likely we'll forget to make sure that the fifth property is actually duties - and errors will surely come right after that